common/src/main/java/com/abrenoch/hyperiongrabber/common/util/BorderProcessor.java

```diff
@@ -0,0 +1,177 @@
+    private BorderObject mCurrentBorder;
+    private int mConsistentFrames = 0;
+    private int mInconsistentFrames = 0;
+
+    public BorderProcessor(int blackThreshold) {
+        BLACK_THRESHOLD = blackThreshold;
+    }
+
+    public BorderObject getCurrentBorder() { return mCurrentBorder; }
+
+    private void checkNewBorder(BorderObject newBorder) {
+        if (mPreviousBorder != null && mPreviousBorder.equals(newBorder)) {
+            ++mConsistentFrames;
+            mInconsistentFrames = 0;
+        } else {
+            ++mInconsistentFrames;
+
+            if (mInconsistentFrames <= MAX_INCONSISTENT_FRAME_COUNT) {
+                return;
+            }
+
+            mPreviousBorder = newBorder;
+            mConsistentFrames = 0;
+        }
+
+        if (mCurrentBorder != null && mCurrentBorder.equals(newBorder)) {
+            mInconsistentFrames = 0;
+            return;
+        }
+
+        if (!newBorder.isKnown()) {
+            if (mConsistentFrames == MAX_UNKNOWN_FRAME_COUNT) {
+                mCurrentBorder = newBorder;
+            }
+        } else {
+            if (mCurrentBorder == null || !mCurrentBorder.isKnown() ||
+                    mConsistentFrames == BORDER_CHANGE_FRAME_COUNT) {
+                        mCurrentBorder = newBorder;
+            }
+        }
+    }
+
+    public void parseBorder(ByteBuffer buffer, int width, int height, int rowStride,
+                            int pixelStride) {
+        checkNewBorder( findBorder(buffer, width, height, rowStride, pixelStride) );
+    }
+
+    private BorderObject findBorder(ByteBuffer buffer, int width, int height, int rowStride,
+                                    int pixelStride) {
+
+        int width33percent = width / 3;
+        int width66percent = width33percent * 2;
+        int height33percent = height / 3;
+        int height66percent = height33percent * 2;
+        int yCenter = height / 2;
+        int xCenter = width / 2;
+        int firstNonBlackYPixelIndex = -1;
+        int firstNonBlackXPixelIndex = -1;
+
+        int c1R, c1G, c1B;
+        int c2R, c2G, c2B;
+        int c3R, c3G, c3B;
+        int pos33p, pos66p;
+
+        buffer.position(0).mark();
+
+        for (int x = 0, posYCLp; x < width33percent; x++) {
+
+            // RGB values at 33% height - to left of image
+            pos33p = (height33percent * rowStride) + (x * pixelStride);
+            c1R = buffer.get(pos33p) & 0xff;
+            c1G = buffer.get(pos33p + 1) & 0xff;
+            c1B = buffer.get(pos33p + 2) & 0xff;
+
+            // RGB values at 66% height - to left of image
+            pos66p = (height66percent * rowStride) + (x * pixelStride);
+            c2R = buffer.get(pos66p) & 0xff;
+            c2G = buffer.get(pos66p + 1) & 0xff;
+            c2B = buffer.get(pos66p + 2) & 0xff;
+
+            // RGB values at center Y - to right of image
+            posYCLp = (yCenter * rowStride) + ((width - x - 1) * pixelStride);
```

Comment:
You lost me here. don't know what CLp means, sorry

Reply:
I think that originally meant Center Left Pixel, but that isn't even accurate anymore because it is reading from the right of the image, not the left. I'll make these a bit less ambiguous